docs/distribution/README.md

```diff
@@ -0,0 +1,528 @@
+
+The following options are offered:
+
+1. [Option 1: Signature Listing API](#signature-discovery---option-1-rest-api-standard-paging)
+1. [Option 2: Generic Reference Listing API](#signature-discovery---option-2-generic-reference-listing-api)
+
+### Signature Discovery - Option 1: Signature Listing API
+
+Similar to the [_tags api][tags-api], a new signatures API is proposed. The signatures API uses the digest as the path to find all signatures related to said digest.
+In the below example, the `net-monitor:v1` tag has a digest of: `sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042`
+
+```HTTP
+GET /v2/<name>/manifests/sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042/signatures/
+```
+
+The response will be in the following format:
+
+```HTTP
+200 OK
+Content-Type: application/json
+{
+  "digest": "sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042",
+  "signatures": [
+    {
+      "digest": "sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042",
+      "mediaType": "application/vnd.oci.image.index.v1+json",
+      "size": "1024",
+      "signature-typ": "x509"
+    },
+    {
+      "digest": "sha256:007170c33ebc4a74a0a554c86ac2b28ddf3454a5ad9cf90ea8cea9f9e75a153b",
+      "mediaType": "application/vnd.oci.image.index.v1+json",
+      "size": "1025",
+      "signature-typ": "x509"
+    }
+  ]
+}
+```
+
+### Signature Discovery - Option 2: Generic Reference Listing API
+
+A slight alternative to the signatures API is to provide a generic reference listing API, where a paged collection of references are returned.
+
+In the below example, the `net-monitor:v1` tag has a digest of: `sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042`
+
+```HTTP
+GET /v2/<name>/manifests/sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042/references/
+```
+
+The response will be in the following format:
+
+```HTTP
+200 OK
+Content-Type: application/json
+{
+  "digest": "sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042",
+  "references": [
+    {
+      "digest": "sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042",
+      "mediaType": "application/vnd.oci.image.index.v1+json",
+      "size": "1024",
+      "config-mediaType": "application/vnd.cncf.notary.config.v2+jwt"
+    },
+    {
+      "digest": "sha256:007170c33ebc4a74a0a554c86ac2b28ddf3454a5ad9cf90ea8cea9f9e75a153b",
+      "mediaType": "application/vnd.oci.image.index.v1+json",
+      "size": "1025",
+      "config-mediaType": "application/vnd.cncf.notary.config.v2+jwt"
+    },
+    {
+      "digest": "sha256:007170c33ebc4a74a0a554c86ac2b28ddf3454a5ad9cf90ea8cea9f9e75a153b",
+      "mediaType": "application/vnd.oci.image.index.v1+json",
+      "size": "1025",
+      "config-mediaType": "application/vnd.oci.image.index.v1+json"
+    }
+  ]
+}
+```
+
+### Paging Results
```

Comment:
From a spec perspective, I'm not sure that we need to prescribe any particular company's design guidelines.  What might be better is making sure that we can specify the minimum necessary such that registries are free to implement pagination however they prefer.  For example, a fairly open way to specify a pagination requirement might be:

* A client can make a request for _N_ entries.  The server may respond with _M_ results, where M <= N.
* When additional results are available, the server should supply a pagination token that can be used in a subsequent request (the `Link: <url>; rel="next"` approach from the distribution spec is a good choice)

Notably, if we don't specify the format of the token (beyond whether it should be a URL, a URL parameter, POSTed data, whatever) registry implementations are free to use an opaque token of their choice.  Implementations are also free to choose things like: do they provide a stable sort order, does the token expire, etc.

Reply:
Agreed. We just need to specify what specific pattern we want to follow. We just pulled up some examples.